Managers/GuildManager.cs

```diff
@@ -133,7 +133,7 @@ public async Task<bool> SyncRolesForGuildUserAsync(IGuildUser user, GuildSetting
                         rolesToAdd.Add(roleId);
                 }
 
-            var rolesToRemove = guildSettings.DataCenterRoleMap.Values.Where(r => !rolesToKeep.Contains(r)).ToHashSet();
+            var rolesToRemove = guildSettings.DataCenterRoleMap.Values.Where(r => !rolesToKeep.Contains(r) && user.RoleIds.Contains(r)).ToHashSet();

```

Comment:
Identified an issue wherein this `rolesToRemove` list would ALWAYS have something in it. It'd list all the roles the user should not have, rather than the roles that the user DOES have but needs removing. This caused the who function to return `true` as if it had made changes to the user's role when it in fact hadn't. 

The bug to the user would have been every member that joined the server receiving the "I've updated your roles" message even though no changed had actually been made and they were just normal users not venue managers.

Reply:
oooh I do remember seeing that more often then it was supposed to happen xD That explains why